jigglypuffRL/deeprl/vpg/vpg.py

```diff
@@ -149,26 +148,64 @@ def create_model(self):
 
 
     def select_action(self, state):
-        state = torch.as_tensor(state).float().to(self.device)
+        state = Variable(torch.as_tensor(state).float().to(self.device))
 
         # create distribution based on policy_fn output
         a, c = self.ac.get_action(state)
-        val = self.ac.get_value(state)
+        val = self.ac.get_value(state).unsqueeze_(0)
 
         # store policy probs and value function for current traj
         self.policy_hist = torch.cat(
             [self.policy_hist, c.log_prob(a).unsqueeze(0)]
         )
 
+        self.value_hist = torch.cat([self.value_hist, val])
+
         # clear traj history
         self.traj_reward = []
         self.policy_hist = Variable(torch.Tensor())
         self.value_hist = Variable(torch.Tensor())
```

Comment:
This should not be done within `select_action`, only in `get_traj_loss()`. No clue how these errors are coming up now

Reply:
Lol, well this is why its not training